### ISSUE TITLE
`@actions/http-client` - unnecessary dependency upon `undici`

### ISSUE THREAD

**Issue by Jason3S** [actions/toolkit#1560]
When updating today, I noticed that size of my action bloated by 15k lines because `undici` was included as a dependency.

`undici` should have been in `devDependencies`, but was accidentally included in `dependencies`.

_Originally posted by @Jason3S in https://github.com/actions/toolkit/pull/1547#discussion_r1360354392_

**Comment by jcesarmobile** [comment]
I don't think it can be a dev dependency since it's imported here
https://github.com/actions/toolkit/blob/fe3e7ce9a7f995d29d1fcfd226a32bca407f9dc8/packages/http-client/src/index.ts#L9

but the undici addition is breaking an action we have, not sure why

**Comment by Jason3S** [comment]
@jcesarmobile,

You are right. I'm still not seeing why it needs to be added to @actions/http-client.

**Comment by manisha1997** [comment]
Do we have any solution to this problem here?

**Comment by foooomio** [comment]
As a workaround, I've decided to downgrade it to v2.1.1.

In the package.json

```
"overrides": {
  "@actions/http-client": "2.1.1"
}
```

**Comment by manisha1997** [comment]
That could be a solution.
I unfortunately depend on actions/core, which depends on actions/http-client.
Do we have a solution for that?

**Comment by foooomio** [comment]
@manisha1997 No problem. Try it.

https://docs.npmjs.com/cli/v11/configuring-npm/package-json#overrides
